File: crm/app/internal/ports/app_user_repository.go
// Package ports defines inbound and outbound interfaces for the CRM hexagonal architecture.
// Nothing in this package may import sqlite, http, or any adapter package.
package ports

import (
	"context"

	"github.com/vantt/data-integration/crm/app/internal/domain"
)

// AppUserRepository is the outbound port for persisting and querying CRM app users.
type AppUserRepository interface {
	GetByEmail(ctx context.Context, email string) (*domain.AppUser, error)
	ListActive(ctx context.Context) ([]domain.AppUser, error)
}

File: crm/src/domain/ports/app_user_repository.py
from __future__ import annotations

from typing import Protocol

from domain.entities.app_user import AppUser


class AppUserRepository(Protocol):
    """Outbound port for persisting and querying CRM app users."""

    def get_by_email(self, email: str) -> AppUser | None:
        """Return a user by email address, or None if not found."""
        ...

    def list_active(self) -> list[AppUser]:
        """Return all active app users."""
        ...

    def update(self, user_id: str, **kwargs: object) -> None:
        """Update one or more columns (email, full_name, role, is_active, ...) for user_id."""
        ...
